examples/cms-sanity/lib/sanity.js

```diff
@@ -1,24 +1,47 @@
-import sanityClient from '@sanity/client'
-import sanityImage from '@sanity/image-url'
+import {
+  createClient,
+  createImageUrlBuilder,
+  createPreviewSubscriptionHook,
+} from 'next-sanity'
 
 const options = {
   // Find your project ID and dataset in `sanity.json` in your studio project
-  dataset: 'production',
+  dataset: process.env.NEXT_PUBLIC_SANITY_DATASET || 'production',
   projectId: process.env.NEXT_PUBLIC_SANITY_PROJECT_ID,
   useCdn: process.env.NODE_ENV === 'production',
   // useCdn == true gives fast, cheap responses using a globally distributed cache.
   // Set this to false if your application require the freshest possible
   // data always (potentially slightly slower and a bit more expensive).
 }
 
-const client = sanityClient(options)
+export const sanityClient = createClient(options)
```

Comment:
Is the client used by the browser bundle? having the client alongside react utils means everything here will be available for both builds, is that intended?

Reply:
If `getStaticProps` and `getStaticPaths` are never included in the browser bundle, it's not needed by the browser bundle. If tree shaking isn't smart enough to get rid of this, I'm not quite sure what the best way to deal with this is, though? Moving it to a separate file doesn't really help, I would assume. Ideas?

Reply:
You'll have to move it into a different file to ensure it can be tree shaken. You can learn more here: https://next-code-elimination.vercel.app/

Reply:
Thanks. Done.